File: backend/week3.js
function Remove(str) {
    let string = {}
    let result = ""
    for (let i of str) {
        if (!string[i]) {
            string[i] = true
            result += i

        }
    }
    return result
}

function IsAnagram(s1, s2) {
    if (s1.length !== s2.length) return false
    freq = {}
    for (let i of s1) {

        freq[i] = (freq[i] | 0) + 1

    }
    for (let i of s2){
        if(!freq[i]){
            return false
        }
        freq[i]--
    }
    return true
}

console.log(Remove("ollweooe"))
console.log(IsAnagram("listen", "stenli"))

File: backend/week3.py
def RemoveStr(str):
    seen=set()
    result=""
    for i in str:
        if i not in seen:
            seen.add(i)
            result+=i
    return result

def IsAnagram(s1, s2):
    if len(s1) != len(s2):
        return False
    freq={}
    for i in s1:
        if i in freq:
            freq[i]+=1
        else:
            freq[i]=1
    for i in s2:
        if i not in freq or freq[i]==0:
            return False
        freq[i]-=1
    return True


print(RemoveStr("hello"))
print(IsAnagram("hello", "llooh"))
